waiting/__init__.py

```diff
@@ -19,12 +19,15 @@ def wait(*args, **kwargs):
 
 
 def iterwait(predicate, timeout_seconds=None, sleep_seconds=1, result=None, waiting_for=None,
-             expected_exceptions=()):
+             on_poll=lambda: None, expected_exceptions=()):
```

Comment:
One more nitpick: It's better to default `on_poll` to `None` than to a no-op lambda. It makes people wonder less when using `help()` or inspecting the arguments...